agent/ecscni/plugin.go

```diff
@@ -110,92 +109,115 @@ func (client *cniClient) SetupNS(cfg *Config) (*current.Result, error) {
 // CleanupNS will clean up the container namespace, including remove the veth
 // pair and stop the dhclient
 func (client *cniClient) CleanupNS(cfg *Config) error {
-	cns := &libcni.RuntimeConf{
+	runtimeConfig := libcni.RuntimeConf{
 		ContainerID: cfg.ContainerID,
 		NetNS:       fmt.Sprintf(netnsFormat, cfg.ContainerPID),
-		IfName:      defaultEthName,
 	}
 
+	os.Setenv("ECS_CNI_LOGLEVEL", logger.GetLevel())
+	defer os.Unsetenv("ECS_CNI_LOGLEVEL")
+	seelog.Debugf("ecscni: Starting clean up the container namespace: %s", cfg.ContainerID)
 	// clean up the network namespace is separate from releasing the IP from IPAM
-	networkConfigList, err := client.createNetworkConfig(cfg, client.createBridgeNetworkConfigWithoutIPAM)
+	err := client.bridgeDel(runtimeConfig, cfg)
 	if err != nil {
-		return errors.Wrap(err, "cni invocation: failed to construct network configuration for namespace cleanup")
+		return errors.Wrap(err, "cni cleanup: invoke bridge plugin failed")
 	}
+	seelog.Debugf("ecscni: bridge cleanup done: %s", cfg.ContainerID)
 
-	seelog.Debugf("Starting clean up the container namespace: %s", cfg.ContainerID)
-	os.Setenv("ECS_CNI_LOGLEVEL", logger.GetLevel())
-	defer os.Unsetenv("ECS_CNI_LOGLEVEL")
-	return client.libcni.DelNetworkList(networkConfigList, cns)
+	err = client.eniDel(runtimeConfig, cfg)
+	if err != nil {
+		return errors.Wrap(err, "cni cleanup: invoke eni plugin failed")
+	}
+	seelog.Debugf("ecscni: container namespace cleanup done: %s", cfg.ContainerID)
+	return nil
 }
 
 // ReleaseIPResource marks the ip available in the ipam db
 func (client *cniClient) ReleaseIPResource(cfg *Config) error {
-	cns := &libcni.RuntimeConf{
+	runtimeConfig := &libcni.RuntimeConf{
 		ContainerID: cfg.ContainerID,
 		NetNS:       fmt.Sprintf(netnsFormat, cfg.ContainerPID),
-		IfName:      defaultEthName,
+		IfName:      defaultVethName,
 	}
 
 	ipamConfig, err := client.createIPAMNetworkConfig(cfg)
 	if err != nil {
 		return err
 	}
 
-	networkConfigList := &libcni.NetworkConfigList{
-		CNIVersion: client.cniVersion,
-		Plugins:    []*libcni.NetworkConfig{ipamConfig},
-	}
-
 	seelog.Debugf("Releasing the ip resource from ipam db, id: [%s], ip: [%v]", cfg.ID, cfg.IPAMV4Address)
 	os.Setenv("ECS_CNI_LOGLEVEL", logger.GetLevel())
 	defer os.Unsetenv("ECS_CNI_LOGLEVEL")
-	return client.libcni.DelNetworkList(networkConfigList, cns)
+	return client.libcni.DelNetwork(ipamConfig, runtimeConfig)
 }
 
-// createNetworkConfig creates the configuration for invoking plugins
-func (client *cniClient) createNetworkConfig(cfg *Config, bridgeConfigFunc func(*Config) (*libcni.NetworkConfig, error)) (*libcni.NetworkConfigList, error) {
-	bridgeConfig, err := bridgeConfigFunc(cfg)
+// bridgeAdd invokes the bridge and ipam plugin to set up the bridge,
+// also it sets the veth pair and route in container namespace
+func (client *cniClient) bridgeAdd(runtimeConfig libcni.RuntimeConf, cfg *Config) (cnitypes.Result, error) {
```

Comment:
Can you combine `bridgeAdd` and `eniAdd` into a single method? I think you can pass `createBridgeNetworkConfigWithIPAM` and `createENINetworkConfig` as parameters to this method